src/odemis/util/filename.py

```diff
@@ -211,10 +211,19 @@ def individualize_filename(fn, names):
     if fn not in names:
         return fn
 
-    # TODO: Detect if filename has counter already
+    # Detect if filename has counter already, otherwise use 1
+    cnt_ptn = r'-\d{1,5}'
+    cnt_m = None
+    for m in re.finditer(cnt_ptn, fn):
+        cnt_m = m
+    # if multiple numbers are present, use last
+    if cnt_m:
+        cnt = int(cnt_m.group()[1:])
+        fn = fn[:cnt_m.start()]
```

Comment:
I think you want the same as in the original code `+ "{cnt}" + fn_ptn[cnt_m.end():]`. Otherwise, as soon as the filename has -number, everything after it is removed. 
Maybe in the else block you can do `fn = fn + "{cnt}"`